### DISCUSSION TITLE
GraphQL API returning a specific file's text at a commit

### DISCUSSION THREAD

**Discussion by benjaminwinokur** [community/community#24465] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I am running into a problem getting a specific text file’s contents at a certain commit using GraphQL</p>
<ol>
<li>
<p>I can get the latest version of the file using the Repository Object</p>
<p>query = {<br>
repository(owner: “” name:""){<br>
object(expression: “:”){<br>
… on Blob{<br>
text<br>
}<br>
}<br>
}<br>
}</p>
</li>
<li>
<p>I can get all the files in the root directory at a certain commit</p>
<p>query = {<br>
repository(owner: “” name:""){<br>
object(oid:""){<br>
… on Commit{<br>
oid<br>
tree{<br>
entries{<br>
name<br>
object{<br>
… on Blob{<br>
text<br>
}<br>
}<br>
}<br>
}<br>
}<br>
}<br>
}<br>
}</p>
</li>
</ol>
<p>But I cannot figure out how to only get the file I am interested in.  Is this possible? And if so, what would the query look like?</p>

**Comment by nickvanw** [comment, marked as answer]
<p>Hi <a class="mention" href="/benjaminwinokur">@benjaminwinokur</a>!</p>
<p>I believe this can be done by mashing up your first and second query by using the OID and the path to the file, something like this:</p>
<pre><code>query {
  repository(owner: "blah", name: "blah") {
    folder: object(expression: "$OID:README.md") {
      ... on Blob {
        text
      }
    }
  }
}
</code></pre>
<p>Using this I was able to get the contents of the README at any commit, let me know if that works for you!</p>
